src/main/scala/se/nimsa/dcm4che/streams/DicomParts.scala

```diff
@@ -92,23 +91,32 @@ object DicomParts {
       val newBytes = header.vr.toBytes(newValue, cs)
       val needsPadding = newBytes.size % 2 == 1
       val newLength = if (needsPadding) newBytes.size + 1 else newBytes.size
-      val updatedHeader = header.withUpdatedLength(newLength.toShort)
-      val updatedValue = if (needsPadding) {
+      val updatedHeader = header.withUpdatedLength(newLength)
+      val updatedValue = if (needsPadding)
         ByteString.fromArray(newBytes :+ header.vr.paddingByte().toByte)
-      } else {
+      else
         ByteString.fromArray(newBytes)
-      }
-      DicomAttribute(updatedHeader, Seq(DicomValueChunk(header.bigEndian, updatedValue, true)))
+      DicomAttribute(updatedHeader, Seq(DicomValueChunk(header.bigEndian, updatedValue, last = true)))
     }
 
     // DA: A string of characters of the format YYYYMMDD, 8 bytes fixed
     def withUpdatedDateValue(newValue: String, cs: SpecificCharacterSet = SpecificCharacterSet.ASCII): DicomAttribute = {
       val newBytes = header.vr.toBytes(newValue, cs)
       val updatedValue = ByteString.fromArray(newBytes)
-      DicomAttribute(header, Seq(DicomValueChunk(header.bigEndian, updatedValue, true)))
+      DicomAttribute(header, Seq(DicomValueChunk(header.bigEndian, updatedValue, last = true)))
     }
 
     def asDicomParts: Seq[DicomPart] = header +: valueChunks
   }
 
+  case object DicomEndMarker extends DicomPart {
+    def bigEndian = false
+    def bytes = ByteString.empty
+  }
+
+  case class DicomAttributes(attributes: Seq[DicomAttribute]) extends DicomPart {
+    def bigEndian = attributes.headOption.exists(_.bigEndian)
+    def bytes = attributes.map(_.bytes).reduce(_ ++ _)
```

Comment:
What's the expected result for DicomAttributes.bytes? As DicomAttribute is implemented right now this will only return the bytes of the ValueChunks, but not bytes from the Headers, not sure if that's waht we want?